src/Serialization.cpp

```diff
@@ -0,0 +1,1310 @@
+    case Split::SplitType::FuseVars:
+        return Halide::Serialize::SplitType::SplitType_FuseVars;
+    case Split::SplitType::PurifyRVar:
+        return Halide::Serialize::SplitType::SplitType_PurifyRVar;
+    default:
+        user_assert(false) << "Unsupported split type\n";
+        return Halide::Serialize::SplitType::SplitType_SplitVar;
+    }
+}
+
+Halide::Serialize::DimType Serializer::serialize_dim_type(const DimType &dim_type) {
+    switch (dim_type) {
+    case DimType::PureVar:
+        return Halide::Serialize::DimType::DimType_PureVar;
+    case DimType::PureRVar:
+        return Halide::Serialize::DimType::DimType_PureRVar;
+    case DimType::ImpureRVar:
+        return Halide::Serialize::DimType::DimType_ImpureRVar;
+    default:
+        user_assert(false) << "Unsupported dim type\n";
+        return Halide::Serialize::DimType::DimType_PureVar;
+    }
+}
+
+Halide::Serialize::LoopAlignStrategy Serializer::serialize_loop_align_strategy(const LoopAlignStrategy &loop_align_strategy) {
+    switch (loop_align_strategy) {
+    case LoopAlignStrategy::AlignStart:
+        return Halide::Serialize::LoopAlignStrategy::LoopAlignStrategy_AlignStart;
+    case LoopAlignStrategy::AlignEnd:
+        return Halide::Serialize::LoopAlignStrategy::LoopAlignStrategy_AlignEnd;
+    case LoopAlignStrategy::NoAlign:
+        return Halide::Serialize::LoopAlignStrategy::LoopAlignStrategy_NoAlign;
+    case LoopAlignStrategy::Auto:
+        return Halide::Serialize::LoopAlignStrategy::LoopAlignStrategy_Auto;
+    default:
+        user_assert(false) << "Unsupported loop align strategy\n";
+        return Halide::Serialize::LoopAlignStrategy::LoopAlignStrategy_AlignStart;
+    }
+}
+
+Halide::Serialize::ExternFuncArgumentType Serializer::serialize_extern_func_argument_type(const ExternFuncArgument::ArgType &extern_func_argument_type) {
+    switch (extern_func_argument_type) {
+    case ExternFuncArgument::ArgType::UndefinedArg:
+        return Halide::Serialize::ExternFuncArgumentType::ExternFuncArgumentType_UndefinedArg;
+    case ExternFuncArgument::ArgType::FuncArg:
+        return Halide::Serialize::ExternFuncArgumentType::ExternFuncArgumentType_FuncArg;
+    case ExternFuncArgument::ArgType::BufferArg:
+        return Halide::Serialize::ExternFuncArgumentType::ExternFuncArgumentType_BufferArg;
+    case ExternFuncArgument::ArgType::ExprArg:
+        return Halide::Serialize::ExternFuncArgumentType::ExternFuncArgumentType_ExprArg;
+    case ExternFuncArgument::ArgType::ImageParamArg:
+        return Halide::Serialize::ExternFuncArgumentType::ExternFuncArgumentType_ImageParamArg;
+    default:
+        user_assert(false) << "Unsupported extern func argument type\n";
+        return Halide::Serialize::ExternFuncArgumentType::ExternFuncArgumentType_UndefinedArg;
+    }
+}
+
+flatbuffers::Offset<flatbuffers::String> Serializer::serialize_string(flatbuffers::FlatBufferBuilder &builder, const std::string &str) {
```

Comment:
Style nit: is it worth putting `using flatbuffers::Offset;` at the top of this file to make all this terser? (Ditto for various other flatbuffers types)